src/jepsen/mongodb/core.clj

```diff
@@ -340,24 +230,44 @@
     (info node "waiting for primary")
     (await-primary conn)
 
-    (info node "primary is" (primary conn))
+    (info node "primary is" (str (primary conn)))
     (jepsen/synchronize test)))
 
 (defn db
   "MongoDB for a particular HTTP URL"
   [clock url]
-  (let [setup-called (atom false)]
+  (let [state (atom {})]
+    ; We intentionally do not implement the jepsen.db/LogFiles protocol in order
+    ; to avoid the jepsen.core/snarf-logs! function from being called when an
+    ; exception occurs or when `db` is being torn down. The
+    ; jepsen.core/snarf-logs! function uses the jepsen.control/download
+    ; function, which requires the use of an SSH connection and is therefore
+    ; incompatible with the concept of running without any virtualization. The
+    ; teardown! method is instead responsible for downloading/copying the logs
+    ; to the store/ directory.
     (reify db/DB
       (setup! [_ test node]
-        (swap! setup-called (constantly true))
+        (swap! state assoc node {:setup-called true})
```

Comment:
Ahhhh, riiggght, now that I see this I understand why you introduced a second protocol for local file downloading. I still don't know what the right way is to handle this--whether we should extend jepsen.db and jepsen.control with the notions of local and remote providers, or what the proper module boundaries might be there... but yeah, this is a bit awkward. Food for thought.